### PR TITLE
Add port error handling to ServerRunner

### PR DESCRIPTION
**Pull request by lebdron**
Signed-off-by: Andrei Lebedev <lebdron@gmail.com>

### Description of the Change
ServerRunner returns Result with port number or error message if server could not start with specified address.

### Benefits
Handle port bind error.
Run ServerRunner with port 0 to auto assign port - IR-741

### Possible Drawbacks 
`reuse` parameter with default is added to ServerRunner constructor. It can be used in components, but now it is used only for tests.

### Usage Examples or Tests *[optional]*
https://github.com/hyperledger/iroha/blob/60646ae7aa6fb9aff6ef857eaeef26446260aefc/test/module/irohad/main/server_runner_test.cpp#L34-L51

### CODE REVIEW

File: test/module/irohad/main/server_runner_test.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      [](iroha::expected::Value<int> x) { return x.value; },
+      [](iroha::expected::Error<std::string> x) { return 0; });
+
+  ServerRunner first_runner((address % 0).str());
+  auto first_query_service = std::make_shared<StubQueryService>();
+  auto result = first_runner.append(first_query_service).run();
+  auto port = boost::apply_visitor(port_visitor, result);
+  ASSERT_NE(0, port);
+
+  ServerRunner second_runner((address % port).str(), false);
```
Comment (luckychess) [inline test/module/irohad/main/server_runner_test.cpp]: May be we could also add a test case for ```reuse=true```?

File: test/module/irohad/torii/torii_queries_test.cpp [hunk 2]
```diff
@@ -28,6 +28,7 @@ limitations under the License.
 #include "model/permissions.hpp"
 #include "torii/processor/query_processor_impl.hpp"
 #include "torii/processor/transaction_processor_impl.hpp"
+#include "torii/command_service.hpp"
```
Comment (luckychess) [inline test/module/irohad/torii/torii_queries_test.cpp]: Is this header have to be here?

File: test/module/irohad/main/server_runner_test.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#include <gtest/gtest.h>
+#include <boost/format.hpp>
+
+#include "endpoint.grpc.pb.h"
+#include "main/server_runner.hpp"
+
+/**
+ * Stub service required to run a server
+ */
+class StubQueryService : public iroha::protocol::QueryService::Service {};
```
Comment (luckychess) [inline test/module/irohad/main/server_runner_test.cpp]: Why not to just use iroha::protocol::QueryService::Service directly?

File: irohad/main/server_runner.cpp [hunk 4]
```diff
@@ -16,35 +16,48 @@
...
+ServerRunner::ServerRunner(const std::string &address, bool reuse)
+    : serverAddress_(address) {
+  if (not reuse) {
+    builder_.AddChannelArgument(GRPC_ARG_ALLOW_REUSEPORT, 0);
+  }
+}
+
+ServerRunner &ServerRunner::append(std::shared_ptr<grpc::Service> service) {
+  builder_.RegisterService(service.get());
+  services_.push_back(service);
```
Comment (luckychess) [inline irohad/main/server_runner.cpp]: May be it would be better to only push services here but register only in run(), what do you think?

Comment (lebdron) [reply irohad/main/server_runner.cpp]: Yes. As @luckychess noted, method name `append` does not imply registering service.

File: test/module/irohad/main/server_runner_test.cpp (hunk 1 above)
Comment (lebdron) [reply test/module/irohad/main/server_runner_test.cpp]: Done.

**Comment by lebdron** [comment]
https://github.com/hyperledger/iroha/commit/0e4128a5407f7c1c565d4a42410254f6ef90ad55

**Pull request closed without merging**